pkg/hubble/container/ring.go

```diff
@@ -31,6 +32,67 @@ var (
 	ErrInvalidRead = errors.New("read position is no longer valid")
 )
 
+// Capacity is the interface that wraps Cap.
+type Capacity interface {
+	// Cap returns the actual capacity.
+	Cap() capacity
+	// AsInt returns the actual capacity as an int.
+	AsInt() int
+}
+
+// capacity implements Capacity.
+type capacity uint16
+
+// Cap returns the actual capacity.
+func (c capacity) Cap() capacity {
+	return c
+}
+
+// AsInt returns the actual capacity as an int.
+func (c capacity) AsInt() int {
+	return int(c)
+}
+
+// CapacityN represent possible buffer capacities for Ring where N is the
+// actual capacity.
+const (
+	Capacity1 capacity = 1<<(iota+1) - 1
+	Capacity3
+	Capacity7
+	Capacity15
+	Capacity31
+	Capacity63
+	Capacity127
+	Capacity255
+	Capacity511
+	Capacity1023
+	Capacity2047
+	Capacity4095
+	Capacity8191
+	Capacity16383
+	Capacity32767
+	Capacity65535
```

Comment:
realistically we probably don't need 1 3 7 15.... they are rather small to have any use whatsoever

Reply:
Capacity 1 and 15 are already used in unit tests and removing some of the capacities from the list would make defining the consts a little less straightforward.